src/Microsoft.Diagnostics.Monitoring.Options/CollectionRuleStateHolder.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Microsoft.Diagnostics.Monitoring.WebApi
+{
+    internal class CollectionRuleStateHolder
+    {
+        public CollectionRuleState CurrState { get; private set; } = CollectionRuleState.Running;
+        public string CurrStateReason { get; private set; } = CollectionRuleStateReasons.Running;
+
+        internal void BeginActionExecution()
+        {
+            CurrState = CollectionRuleState.ActionExecuting;
```

Comment:
Would it make sense to error check previous states? For example, it might not make sense to transition from Finished to Running again.

Reply:
Debug.Assert (will only be for our own internal purposes)